optimizely/event_builder.py

```diff
@@ -35,78 +35,90 @@ class BaseEventBuilder(object):
 
   def __init__(self, config):
     self.config = config
-    self.params = {}
 
   @abstractproperty
   class EventParams(object):
     pass
 
-  def _add_project_id(self):
-    """ Add project ID to the event. """
+  def _get_project_id(self):
+    """ Get project ID.
 
-    self.params[self.EventParams.PROJECT_ID] = self.config.get_project_id()
+    Returns:
+      Project ID of the datafile.
+    """
 
-  def _add_account_id(self):
-    """ Add account ID to the event. """
+    return self.config.get_project_id()
 
-    self.params[self.EventParams.ACCOUNT_ID] = self.config.get_account_id()
+  def _get_account_id(self):
+    """ Get account ID.
 
-  @abstractmethod
-  def _add_user_id(self, user_id):
-    """ Add user ID to the event.
-
-    Args:
-      user_id: ID of the user.
+    Returns:
+      Account ID in the datafile.
     """
-    pass
+
+    return self.config.get_account_id()
 
   @abstractmethod
-  def _add_attributes(self, attributes):
-    """ Add attribute(s) information to the event.
+  def _get_attributes(self, attributes):
+    """ Get attribute(s) information.
 
     Args:
       attributes: Dict representing user attributes and values which need to be recorded.
     """
     pass
 
   @abstractmethod
-  def _add_source(self):
-    """ Add source information to the event. """
+  def _get_source(self):
+    """ Get source information. """
     pass
 
-  def _add_revision(self):
-    """ Add datafile revision information to the event. """
-    pass
+  def _get_anonymize_ip(self):
+    """ Get IP anonymization bool
 
-  def _add_anonymize_ip(self):
-    """ Add IP anonymization bool to the event """
+    Returns:
+      bool 'anonymizeIP' value in the datafile.
+    """
 
-    self.params[self.EventParams.ANONYMIZE_IP] = self.config.get_anonymize_ip_value()
+    return self.config.get_anonymize_ip_value()
 
   @abstractmethod
   def _get_time(self):
-    """ Get time in milliseconds to be added to the event.
+    """ Get time in milliseconds to be added.
 
     Returns:
-      Current time in milliseconds.
+      int Current time in milliseconds.
     """
 
     return int(round(time.time() * 1000))
 
-  def _add_common_params(self, user_id, attributes):
-    """ Add params which are used same in both conversion and impression events.
+  def _get_common_params(self, user_id, attributes):
+    """ Get params which are used same in both conversion and impression events.
 
     Args:
       user_id: ID for user.
       attributes: Dict representing user attributes and values which need to be recorded.
+
+    Returns:
+     dict Common params required for both impression and conversion events.
```

Comment:
Change this to `Dict consisting of parameters common to both impression and conversion events.`